src/Microsoft.AspNetCore.DataProtection.Extensions/DataProtectionProvider.cs

```diff
@@ -40,21 +49,25 @@ public static IDataProtectionProvider Create(DirectoryInfo keyDirectory)
             {
                 throw new ArgumentNullException(nameof(keyDirectory));
             }
-
             if (setupAction == null)
             {
                 throw new ArgumentNullException(nameof(setupAction));
             }
 
+            return CreateProvider(keyDirectory, setupAction);
+        }
+
+        private static IDataProtectionProvider CreateProvider(
+            DirectoryInfo keyDirectory,
+            Action<IDataProtectionBuilder> setupAction)
+        {
             // build the service collection
             var serviceCollection = new ServiceCollection();
             var builder = serviceCollection.AddDataProtection();
+
             builder.PersistKeysToFileSystem(keyDirectory);
```

Comment:
Like @blowdart, I'm a bit surprised this method doesn't throw if the parameter is null. Anyway, I'd personally add a little `if` here, to ensure it's never called when `keyDirectory` is null.